### PR TITLE
Extract scheduler tests into smaller ones

### PR DESCRIPTION
**Pull request by marcusolsson**
The scheduler test is quite massive and personally I had some difficulty of keeping track of the state in each test case. Splitting the test (hopefully) makes it easier to follow.

Also, since Eremetic is using GoConvey for tests, which AFAIK made for BDD-style tests, I took the liberty of reformulating the test cases in the form of Given, When, Then.

The actual tests haven't been modified though. Let me know if you think that's the case or whether my test descriptions are accurate or not.

### CODE REVIEW

File: scheduler/scheduler_test.go [hunk 1]
```diff
@@ -48,337 +51,479 @@ func TestScheduler(t *testing.T) {
...
+	dir, _ := os.Getwd()
+	db, err := database.NewDB("boltdb", fmt.Sprintf("%s/../db/test.db", dir))
+	if err != nil || db == nil {
+		t.Error("Foo")
+		t.Fail()
+	}
+	db.Clean()
+	defer db.Close()
+
+	Convey("Given a scheduler with one scheduled task", t, func() {
```
Comment (keis) [inline scheduler/scheduler_test.go]: BDD formulation nitpick, the given state has not scheduled tasks here

Comment (marcusolsson) [inline scheduler/scheduler_test.go]: Good catch. I believe the copy-paste-devil paid me a visit :+1:

**Comment by keis** [comment]
:+1:

**Comment by marcusolsson** [comment]
@alde explained that the tests are meant to replicate the RSpec style so I've reverted some of the changes to keep that style. Personally, I think that when in Rome, do as the Romans do, but as long as we're consistent, I'm OK with it.

**Comment by alde** [comment]
I'm open to rewriting the tests, but then we should do one branch overhauling all the tests to not be stuck with different styles.

Maybe as a collaboration.

**Pull request merged**